ampligraph/latent_features/models/RandomBaseline.py

```diff
@@ -38,123 +37,110 @@ class RandomBaseline(EmbeddingModel):
 
     def __init__(self, seed=constants.DEFAULT_SEED):
         """Initialize the model
-        
+
         Parameters
         ----------
         seed : int
             The seed used by the internal random numbers generator.
-            
-        """
-        self.seed = seed
-        self.is_fitted = False
-        self.is_filtered = False
-        self.sess_train = None
-        self.sess_predict = None
-
-        self.rnd = check_random_state(self.seed)
-        self.eval_config = {}
 
-    def _fn(e_s, e_p, e_o):
-        pass
-
-    def get_embeddings(self, entities, type='entity'):
-        """Get the embeddings of entities or relations.
+        """
+        super().__init__(k=1, eta=1, epochs=1, batches_count=1, seed=seed, verbose=False)
 
-        .. Note ::
-            Use :meth:`ampligraph.utils.create_tensorboard_visualizations` to visualize the embeddings with TensorBoard.
+    def _fn(self, e_s, e_p, e_o):
+        """Random baseline scoring function: random number between 0 and 1.
 
         Parameters
         ----------
-        entities : array-like, dtype=int, shape=[n]
-            The entities (or relations) of interest. Element of the vector must be the original string literals, and
-            not internal IDs.
-        type : string
-            If 'entity', will consider input as KG entities. If `relation`, they will be treated as KG predicates.
+        e_s : Tensor, shape [n]
+            The embeddings of a list of subjects.
+        e_p : Tensor, shape [n]
+            The embeddings of a list of predicates.
+        e_o : Tensor, shape [n]
+            The embeddings of a list of objects.
 
         Returns
         -------
-        embeddings : None
-            Returns None as this model does not have any embeddings.
-            While scoring, it creates a random score for a triplet.
+        score : TensorFlow operation
+            Random number between 0 and 1.
 
         """
-        return None
+        # During training TensorFlow requires that gradients with respect to the trainable variables exist
+        if self.train_dataset_handle is not None:
+            # Sigmoid reaches 1 quite quickly, so the `useless` variable below is 0 for all practical purposes
+            useless = tf.sigmoid(tf.reduce_mean(tf.clip_by_value(e_s, 1e10, 1e11))) - 1.0
```

Comment:
Because this is a bit arbitrary, it makes me wonder if there is a cleaner way to solve this downstream. I guess overriding the fit() function defeats the purpose of the refactor, and adding another try/except to EmbeddingModel's fit() function isn't much cleaner. Do you have any ideas?

Reply:
I do agree it is not clean, I spent some time considering alternatives but could not get anything better than this hack. However, I don't see any potential problems in practice. Do you see any?

Overriding fit would require adding code to create some internal variables, e.g. rel_to_idx. This is indeed not what we want, as changes to the base class fit would require changes here as well. 

I would rather also not interfere with the fit code in EmbeddingModel.

Reply:
Doing this actually works:

```python
        try:
            super().fit(X, early_stopping, early_stopping_params)
        except ValueError:
            pass
```

However, I don't prefer this solution because of the generic ValueError. It could catch exceptions unrelated to this particular problem. Also, this implicitly expects that aborting in the middle of the `fit` will not leave the object in a bad state, which could change in the future.

Reply:
I don't see any practical problems either, it just confuses the reader for a second when looking at the code because it is, as you said, a hack. 

In any case it is much better than the buggy and hard-to-maintain current version, and the comment sufficiently explains the need for the hack. All is fine for now if you ask me!